### ISSUE TITLE
The second order gradient of `lax.pow` is wrong

### ISSUE THREAD

**Issue by Doraehhh** [google/jax#12033]
The second order gradient of `lax.pow` is wrong. For instance, the `fn` computes `x^y`, so `grad_fn` returns `(y*x^(y-1), ln(x)* x^y)`. When `x, y = 23.0, 0.0`, for the second element `ln(x) * x^y`, the gradient w.r.t `x` should be `1/x` instead of `23.0` computed by `jacrev` or `jacfwd`. 

```python
import jax
def grad_fn(x, y):
    def fn(x, y):
        return jax.numpy.sum(jax.lax.pow(x, y))
    return jax.grad(fn, (0, 1))(x, y)
x = jax.numpy.array([23.0])
y = jax.numpy.array([0.0])

print(jax.jacrev(grad_fn, (0, 1))(x, y))
print(jax.jacfwd(grad_fn, (0, 1))(x, y))
```

```
((DeviceArray([[0.]], dtype=float32), DeviceArray([[23.]], dtype=float32)), (DeviceArray([[0.04347826]], dtype=float32), DeviceArray([[9.831324]], dtype=float32)))
```

**Comment by jakevdp** [comment]
Thanks for the report – I had a bit of trouble following your reproduction, so I rewrote it a bit more succinctly:
```python
import jax

def f(x, y):
  return jax.lax.pow(x, y)

x = 23.0
y = 0.0

print(f"d2f/dxdy = {jax.grad(jax.grad(f, 0), 1)(x, y)}")
print(f"d2f/dydx = {jax.grad(jax.grad(f, 1), 0)(x, y)}")
```
```
d2f/dxdy = 23.0
d2f/dydx = 0.043478261679410934
```
As you mention, these two values should be equal

Looking into the code, I think the issue comes from the `select` statement in line 2004 here: https://github.com/google/jax/blob/da4e79a6251ae71504061b496ea34f20dc06fa60/jax/_src/lax/lax.py#L2003-L2008

The gradient of `pow` with respect to `x` is special-cased when `y` is zero. I think this yields the correct result when `y` is a constant, but it removes the dependence of the result on `y` and therefore the mixed second derivative is incorrect.
